hilti/toolchain/src/compiler/validator.cc

```diff
@@ -139,10 +139,14 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
     // Ensures that the node represented by tag is allowed to have all of the
     // provided attributes. This does not use any context, if more information
     // is needed, then do the check elsewhere.
-    void checkNodeAttributes(node::Tag tag, AttributeSet* attributes, const std::string_view& where) {
+    void checkNodeAttributes(Node* n, node::Tag tag, AttributeSet* attributes, const std::string_view& where) {
```

Comment:
Would it be reasonable to now use `n->nodeTag()` and remove the `tag` from this function?